pkg/virt-controller/watch/vmi/vmi.go

```diff
@@ -178,6 +179,8 @@ type annotationsGenerator interface {
 
 type statusUpdater func(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) error
 
+type validateSpec func(*k8sfield.Path, *virtv1.VirtualMachineInstanceSpec, *virtconfig.ClusterConfig) []v1.StatusCause
```

Comment:
Please consider using a network specific interface, as there is currently no need to generalize the integration point:
```go
type netCreationValidator interface {
  ValidateCreation() []metav1.StatusCause
}
```

Reply:
On second thought, the network validator is not ready to be injected in its current state.

Reply:
I suggest resolving this thread.